watchtower/blob/justice_kit_test.go

```diff
@@ -212,3 +216,195 @@ func testBlobJusticeKitEncryptDecrypt(t *testing.T, test descriptorTest) {
 			"want: %v, got %v", boj, boj2)
 	}
 }
+
+// TestJusticeKitRemoteWitnessConstruction tests that a JusticeKit returns the
+// proper to-remote witnes script and to-remote witness stack. This should be
+// equivalent to p2wkh spend.
+func TestJusticeKitRemoteWitnessConstruction(t *testing.T) {
+	// Generate the to-remote pubkey.
+	toRemotePrivKey, err := btcec.NewPrivateKey(btcec.S256())
+	if err != nil {
+		t.Fatalf("unable to generate to-remote priv key: %v", err)
+	}
+
+	// Copy the to-remote pubkey into the format expected by our justice
+	// kit.
+	var toRemotePubKey blob.PubKey
+	copy(toRemotePubKey[:], toRemotePrivKey.PubKey().SerializeCompressed())
+
+	// Sign a message using the to-remote private key. The exact message
+	// doesn't matter as we won't be validating the signature's validity.
+	digest := bytes.Repeat([]byte("a"), 32)
+	rawToRemoteSig, err := toRemotePrivKey.Sign(digest)
+	if err != nil {
+		t.Fatalf("unable to generate to-remote signature: %v", err)
+	}
+
+	// Convert the DER-encoded signature into a fixed-size sig.
+	commitToRemoteSig, err := lnwire.NewSigFromSignature(rawToRemoteSig)
+	if err != nil {
+		t.Fatalf("unable to convert raw to-remote signature to "+
+			"Sig: %v", err)
+	}
+
+	// Populate the justice kit fields relevant to the to-remote output.
+	justiceKit := &blob.JusticeKit{
+		CommitToRemotePubKey: toRemotePubKey,
+		CommitToRemoteSig:    commitToRemoteSig,
+	}
+
+	// Now, compute the to-remote witness script returned by the justice
+	// kit.
+	toRemoteScript, err := justiceKit.CommitToRemoteWitnessScript()
+	if err != nil {
+		t.Fatalf("unable to compute to-remote witness script: %v", err)
+	}
+
+	// Assert this is exactly the to-remote, compressed pubkey.
+	if !bytes.Equal(toRemoteScript, toRemotePubKey[:]) {
+		t.Fatalf("to-remote witness script should be equal to "+
+			"to-remote pubkey, want: %x, got %x",
+			toRemotePubKey[:], toRemoteScript)
+	}
+
+	// Next, compute the to-remote witness stack, which should be a p2wkh
```

Comment:
The stack for `p2wkh` should be: the sig, and then the pubkey.

Reply:
see below comment

Reply:
Even with the below, p2wkh doesn't have a witness script as its last item. Instead it always has the sig+pubkey combo. Technically, the witness script is actually the full unrolled p2pkh type script.